overlord/snapstate/handlers.go

```diff
@@ -528,21 +548,16 @@ func maybeRestart(t *state.Task, info *snap.Info) {
 	st := t.State()
 	if release.OnClassic && info.Type == snap.TypeOS {
 		t.Logf("Requested daemon restart.")
-		st.Unlock()
 		st.RequestRestart(state.RestartDaemon)
```

Comment:
This one might be a bit problematic. This will a stop/start command, and block for possibly quite a while until it determines for thing to be fine.